cca_test.go

```diff
@@ -0,0 +1,171 @@
+		epsilon   float64
+	}{
+		// Test results verified using R.
+		{ // Truncated iris data, Sepal vs Petal measurements.
+			xdata: mat64.NewDense(10, 2, []float64{
+				5.1, 3.5,
+				4.9, 3.0,
+				4.7, 3.2,
+				4.6, 3.1,
+				5.0, 3.6,
+				5.4, 3.9,
+				4.6, 3.4,
+				5.0, 3.4,
+				4.4, 2.9,
+				4.9, 3.1,
+			}),
+			ydata: mat64.NewDense(10, 2, []float64{
+				1.4, 0.2,
+				1.4, 0.2,
+				1.3, 0.2,
+				1.5, 0.2,
+				1.4, 0.2,
+				1.7, 0.4,
+				1.4, 0.3,
+				1.5, 0.2,
+				1.4, 0.2,
+				1.5, 0.1,
+			}),
+			wantCorrs: []float64{0.7250624174504773, 0.5547679185730191},
+			wantpVecs: mat64.NewDense(2, 2, []float64{
+				0.0765914610875867, 0.9970625597666721,
+				0.9970625597666721, -0.0765914610875868,
+			}),
+			wantqVecs: mat64.NewDense(2, 2, []float64{
+				0.3075184850910837, 0.9515421069649439,
+				0.9515421069649439, -0.3075184850910837,
+			}),
+			wantphiVs: mat64.NewDense(2, 2, []float64{
+				-1.9794877596804641, 5.2016325219025124,
+				4.5211829944066553, -2.7263663170835697,
+			}),
+			wantpsiVs: mat64.NewDense(2, 2, []float64{
+				-0.0613084818030103, 10.8514169865438941,
+				12.7209032660734298, -7.6793888180353775,
+			}),
+			epsilon: 1e-12,
+		},
+		// Test results compared to those results presented in examples by
+		// Koch, Inge. Analysis of multivariate and high-dimensional data.
+		// Vol. 32. Cambridge University Press, 2013. ISBN: 9780521887939
+		{ // ASA Car Exposition Data of Ramos and Donoho (1983)
+			xdata:     carData.View(0, 0, 392, 3), // Displacement, Horsepower, Weight
+			ydata:     carData.View(0, 3, 392, 2), // Acceleration, MPG
+			wantCorrs: []float64{0.8782187384352336, 0.6328187219216761},
+			wantpVecs: mat64.NewDense(3, 2, []float64{
+				0.3218296374829181, 0.3947540257657075,
+				0.4162807660635797, 0.7573719053303306,
+				0.8503740401982725, -0.5201509936144236,
+			}),
+			wantqVecs: mat64.NewDense(2, 2, []float64{
+				-0.5161984172278830, -0.8564690269072364,
+				-0.8564690269072364, 0.5161984172278830,
+			}),
+			wantphiVs: mat64.NewDense(3, 2, []float64{
+				0.0025033152994308, 0.0047795464118615,
+				0.0201923608080173, 0.0409150208725958,
+				-0.0000247374128745, -0.0026766435161875,
+			}),
+			wantpsiVs: mat64.NewDense(2, 2, []float64{
+				-0.1666196759760772, -0.3637393866139658,
+				-0.0915512109649727, 0.1077863777929168,
+			}),
+			epsilon: 1e-12,
+		},
+		// Test results compared to those results presented in examples by
+		// Koch, Inge. Analysis of multivariate and high-dimensional data.
+		// Vol. 32. Cambridge University Press, 2013. ISBN: 9780521887939
+		{ // Boston Housing Data of Harrison and Rubinfeld (1978)
+			xdata:     bostonData.View(0, 0, 506, 7), // Per capita crime rate by town, Proportion of non-retail business acres per town, Nitric oxide concentration (parts per 10 million), Weighted distances to Boston employment centres, Index of accessibility to radial highways, Pupil-teacher ratio by town, Proportion of blacks by town
```

Comment:
Please split these two comments over multiple lines and place them above the field declarations.